script.js

```diff
@@ -1,4 +1,128 @@
+//define vars
+var indivisualDiceRoll;
+var mode = `instructions`;
+var playerNumber = 1;
+var allValues = [];
+var numberOfPlayer = 2;
+var diceNumber = 2;
+var whoPlays = `player`;
+var winCondition = `highest value wins`;
+var diceNumber = 2;
+var allDiceRoll = [];
+var currentDice = 1;
+var category;
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (input == `instructions`) {
+    return instruction();
+  }
+  // if (input == `computer` || input == `player`) {
+  //   whoPlays = input;
+  //   return `Your player has been set to ${input}.Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+
+  if (input == `highest value wins` || input == `lowest value wins`) {
+    winCondition = input;
+    return `Your winning criteria has been set to ${input}.Press submit to start the game. For any queries, enter 'instructions' and press submit.`;
+  }
+  if (mode == `instructions`) {
+    mode = `roll dice`;
+    return instruction();
+  }
+  if (input == `player number` || input == `dice number`) {
+    mode = input;
+
+    return `Your mode has been set to ${input}. Please input the ${input} you want`;
+  }
+  if (mode == `player number`) {
+    numberOfPlayer = input;
+    mode = `instructions`;
+    return `The number of players have been set to ${numberOfPlayer}. Please press submit to start the game. `;
+  }
+  if (mode == `dice number`) {
+    diceNumber = input;
+    mode = `instructions`;
+    return `The number of dice have been set to ${diceNumber}. Please press submit.`;
+  }
+  console.log(1);
+  while (playerNumber <= numberOfPlayer) {
+    console.log(2);
+    if (mode == `roll dice`) {
+      console.log(3);
+      while (currentDice <= diceNumber) {
+        indivisualDiceRoll = diceRoll();
+        allDiceRoll.push(indivisualDiceRoll);
+        console.log(4);
+        currentDice = currentDice + 1;
+      }
+      console.log(5);
+      mode = `select order`;
+      myOutputValue = diceRollMessage(playerNumber);
+      allDiceRoll = [];
+      currentDice = 1;
+      return myOutputValue;
+    }
+    if (mode == `select order`) {
+      console.log(diceNumber, allDiceRoll);
+      allValues.push(Number(input));
+      console.log(allValues);
+      myOutputValue = selectOrderMessage(playerNumber, input);
+      mode = `roll dice`;
+      playerNumber = playerNumber + 1;
+    }
+  }
+  if (winCondition == `highest value wins`) {
+    myOutputValue = `The winner is player ${
+      locationOfHighestValue(allValues) + 1
+    } with a value of ${findHighestValueInArray(allValues)}`;
+  }
+  if (winCondition == `lowest value wins`) {
+    myOutputValue = `The winner is player ${
+      locationOfLowestValue(allValues) + 1
+    } with a value of ${findLowestValueInArray(allValues)}`;
+  }
```

Comment:
Over here, you could consider creating a helper function that returns you a message depending on the win condition. E.g. winningMessage(winCondition).